### PR TITLE
Issue #1453: Exact QUANTILE LIST

### PR DESCRIPTION
**Pull request by hawkfish**
Add a `QUANTILE` variant that takes a `LIST` of parameters and returns a `LIST` of the corresponding quantiles.

### CODE REVIEW

File: test/sql/aggregate/aggregates/test_quantile_list.test [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# name: test/sql/aggregate/aggregates/test_quantile_list.test
+# description: Test QUANTILE operator with LIST quantiles
+# group: [aggregates]
+
+statement ok
+PRAGMA enable_verification
+
+# scalar quantiles
+statement ok
+create table quantiles as select range r, random() from range(10000) union all values (NULL, 0.1), (NULL, 0.5), (NULL, 0.9) order by 2;
```
Comment (Mytherin) [inline test/sql/aggregate/aggregates/test_quantile_list.test:10]: Could you add a tests for quantiles on constant values (e.g. `SELECT quantile(1, LIST_VALUE(0.1, 0.5, 0.9)) FROM quantiles`)?

File: test/sql/aggregate/aggregates/test_quantile_list.test [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
```
Comment (Mytherin) [inline test/sql/aggregate/aggregates/test_quantile_list.test:11]: Same with quantiles on an empty set (`SELECT quantile(r, LIST_VALUE(0.1, 0.5, 0.9)) FROM quantiles WHERE 1=0`)

File: test/sql/aggregate/aggregates/test_quantile_list.test [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+query I
```
Comment (Mytherin) [inline test/sql/aggregate/aggregates/test_quantile_list.test:12]: And an empty list as input (`SELECT quantile(r, LIST_VALUE()) FROM quantiles`)

File: test/sql/aggregate/aggregates/test_quantile_list.test [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+SELECT quantile(r, LIST_VALUE(0.1, 0.5, 0.9)) from quantiles
```
Comment (Mytherin) [inline test/sql/aggregate/aggregates/test_quantile_list.test:13]: I'm also missing a grouped quantile test (i.e. using the quantile with a `GROUP BY` clause present).

File: src/function/aggregate/holistic/quantile.cpp [hunk 5]
```diff
@@ -93,9 +97,11 @@ struct QuantileOperation {
 		D_ASSERT(bind_data_p);
 		auto bind_data = (QuantileBindData *)bind_data_p;
 		auto v_t = (T *)state->v;
-		auto offset = (idx_t)((double)(state->pos - 1) * bind_data->quantile);
-		std::nth_element(v_t, v_t + offset, v_t + state->pos);
-		target[idx] = v_t[offset];
+		for (const auto &quantile : bind_data->quantiles) {
```
Comment (Mytherin) [inline src/function/aggregate/holistic/quantile.cpp:100]: This is always supposed to be a list of size 1, right? Perhaps assert that and explicitly only use the first entry?

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! Looks great. Some comments:

File: test/sql/aggregate/aggregates/test_quantile_list.test (hunk 4 above)
Comment (hawkfish) [reply test/sql/aggregate/aggregates/test_quantile_list.test:13]: Ah, there weren't any in the original test case (which I just cloned).

**Comment by hawkfish** [comment]
Hmm, this didn't fail in my repository...plus I develop on OSX!

**Comment by hawkfish** [comment]
Is there a way to update the [published documentation](https://duckdb.org/docs/sql/aggregates) so that when a release happens, the docs are already up to date?

**Comment by Mytherin** [comment]
> Hmm, this didn't fail in my repository...plus I develop on OSX!

OSX builds on GH actions tend to be a bit flaky in my experience, fixed now.

> Is there a way to update the [published documentation](https://duckdb.org/docs/sql/aggregates) so that when a release happens, the docs are already up to date?

The documentation is in a separate repo, [here](https://github.com/cwida/duckdb-web). Feel free to send a PR there as well!

**Comment by hawkfish** [comment]
Found it - thanks!

**Comment by hawkfish** [comment]
Not sure about the optimiser suggestion. I think it is a good idea, but not something I can um bill to a client(!)

**Pull request merged**